### ISSUE TITLE
Add Hezar to Hugging Face libraries

### ISSUE THREAD

**Issue by arxyzan** [hezarai/hezar#56]
There are a lot of libraries supported in the Hugging Face Hub. ([full list](https://huggingface.co/docs/hub/models-libraries)).
In order to add Hezar to this list we have to take the following steps:
The full guide is [here](https://huggingface.co/docs/hub/models-adding-libraries#register-your-libraries-supported-tasks-on-the-hub) but these are the main steps to do.
- [x] Add upload/download feature to the libraries modules so that everything can be hosted on the Hugging Face Hub. ([guide]())
- [x] Create an Inference API Docker image like [these](https://github.com/huggingface/api-inference-community/tree/main/docker_images) 
- [x] Add it to [the Hugging Face API Inference Community repo](https://github.com/huggingface/api-inference-community/tree/main/docker_images) and send a pull request. ([guide](https://huggingface.co/docs/hub/models-adding-libraries#create-an-inference-api-docker-image))
- [ ] Register supported tasks. ([guide](https://huggingface.co/docs/hub/models-adding-libraries#register-your-libraries-supported-tasks-on-the-hub))

@pooya-mohammadi

**Comment by arxyzan** [comment]
Alright, we have a pretty annoying issue here that we don't want to just fix with a workaround. 

**Problem**
HuggingFace Hub Inference API needs a specific output structure for each of the tasks. Most libraries handle this by converting their model/pipeline outputs to HF format right in their fork of the HF inference API repo. But that has some flaws in my opinion. Schemas in libraries change all the time and you cannot be sure that the output format for the task X is going to be the same in the future. And any change in the models' output schemas means requiring a new PR to the HF api inference community repo which we don't want.

**Possible solutions**
1. We can just convert all of our model output formats to a HF inference API compatible format and we'll probably never send another PR to that repo. 
2. Hezar has a `ModelOutput` (data)class and for each task there is one `<TASK>Output` (data)class. we can add a `to_inference_api_format()` method to each of them that converts Hezar outputs to HF Inference API format. This way we won't ever need to change Hezar model output formats. Besides, if any of the HF Inference API outputs change, we can still adapt to it by changing our `to_inference_api_format()` methods not the codes in the HF inference API repo. Although, this solution has its own issues; Some pipeline outputs like text classification, requires the top 5 best results. Our `TextClassificationOutput` class outputs only the top 1 results for each sample in the batch. So the outputs of a Hezar text classficiation model cannot be converted to the HF format out of the box but we do need logits to do that.
3. Auxiliary pipeline output functions: These functions take in the minimum required parameters to construct the outputs for each task. For example `text_classification_output()` takes in `logits`, `probs`, `id2label` and a `format` parameter (`"hezar"` or `"hf_inference_api"`) and constructs the outputs based on that.

I'm still not fully satisfied with any of the above solutions. I think we can still think of a better one.

Let me know what you think @pooya-mohammadi

**Comment by arxyzan** [comment]
PR sent!🚀
